lib/logstash/util/java_version.rb

```diff
@@ -0,0 +1,52 @@
+module LogStash::Util::JavaVersion
+  # Print a warning if we're on a bad version of java
+  def self.warn_on_bad_java_version
+    if self.bad_java_version?(self.version)
+      STDERR.puts("!!! Please upgrade your java version, the current version '#{self.version}' may cause problems. We recommend a minimum version of 1.7.0_51")
+    end
+  end
+
+  # Check to see if this is a recommended java version, print a warning to stdout if this is a bad version
+  # Returns the current java version
+  def self.version
+    return nil unless LogStash::Environment.jruby?
```

Comment:
I think this could go either way on this. void is a concept from languages that aren't ruby, and are used to denote functions with side effects. In ruby a nil means nil, which is a much more expansive concept than void. Raising an exception here is a stylistic choice in my opinion, but I don't see a strong case for it. If getting a nil java version on MRI makes sense to me, because in that case the java version non-existent AKA nil.